src/lib/crypto/ecdsa.c

```diff
@@ -79,14 +79,9 @@ pgp_ecdsa_sign_hash(rng_t *                 rng,
         goto end;
     }
 
-    // Allocate memory and copy results
+    // Allocate memory and copy results. bn_bin2bn won't return NULL
     sign->r = bn_bin2bn(out_buf, curve_order, sign->r);
     sign->s = bn_bin2bn(out_buf + curve_order, curve_order, sign->s);
-    if (!sign->r || !sign->s) {
```

Comment:
I don't get the point of removing these checks, it can certainly return NULL.

Reply:
Only if sign->r is NULL. But it this is the case then botan_pk_op_sign_finish() will fail.

Reply:
I mean that `bn_bin2bn` can fail if memory allocation fails.

Reply:
I think in this case there is no memory allocation done (out_buf != NULL && sign->r!=NULL)

Reply:
`botan_mp_from_bin` can throw `std::bad_alloc` here https://github.com/randombit/botan/blob/8f17d65056903ca4da9bb16a85db717e4bea4456/src/lib/math/bigint/bigint.cpp#L284, in which case `bn_bin2bn` would return NULL.

Reply:
If exception is thrown then FFI returns BOTAN_FFI_ERROR_EXCEPTION_THROWN, so this check wouldn't work anyway.

Reply:
In such case botan returns BOTAN_FFI_ERROR_EXCEPTION_THROWN, isn't?

Reply:
Yes, and then `bn_bin2bn` would return NULL here: https://github.com/riboseinc/rnp/blob/33fc010a2e2ff4cf08a8eec92159ad9d004d7352/src/lib/crypto/bn.c#L60

Reply:
On a side note, I think it's bad practice to ignore return values from botan's FFI. Just because the current implementation of a function can't fail (not the case here), doesn't mean it won't evolve and introduce some failure points later on.

Reply:
argh.. I'm wrong. sign->r always == NULL, that's even checked above.